Lib/tempfile.py

```diff
@@ -685,6 +685,9 @@ def __exit__(self, exc, value, tb):
     def __iter__(self):
         return self._file.__iter__()
 
+    def __del__(self):
```

Comment:
This shouldn't be added: deleting the SpooledTemporaryFile will null out the reference to the underline file, and therefor call its __del__

Reply:
I think if you inherit the default `IOBase.__del__` implementation, it will call _close_ and defeat any ResourceWarning that might otherwise be emitted. Perhaps it is better to make `__del__` do nothing, or set it to `object.__del__`.

Reply:
I agree: the underlying file should not be explicitly deleted as this is not expected behaviour. I can imagine a situation where someone is deliberately holding a reference to the underlying file and they would not expect/want it to be closed until their own reference has fallen out of scope. I've changed the method to do nothing and added a docstring to reflect this.

Thanks for your feedback

Reply:
> I agree: the underlying file should not be explicitly deleted as this is not expected behaviour

The doc says: """This function operates exactly as TemporaryFile() does, except [irrelevant differences]."""

And then, about TemporaryFile: """On completion of the context or destruction of the file object the temporary file will be removed from the filesystem."""

So it seems the underlying file _should_ be deleted when the file object disappears.

Reply:
The point is that `IOBase` implements a `__del__` which has some side-effects. Those side effects are not desirable here. Any underlying buffers being wrapped, e.g. the `TemporaryFile` or the `BytesIO`, will be gc'd and handled as they should.

I think this implementation is correct.